src/lib/ecosystems/resolve-test-facts.ts

```diff
@@ -40,16 +40,17 @@ export async function resolveAndTestFacts(
   },
   options: Options & PolicyOptions,
 ): Promise<[TestResult[], string[]]> {
-  const unmanagedDepsOverride = process.env.USE_UNMANAGED_DEPS;
+  try {
+    return await resolveAndTestFactsUnmanagedDeps(scans, options);
+  } catch (error) {
+    const unauthorized = error.code === 401 || error.code === 403;
 
-  const featureFlagEnabled = await hasFeatureFlag(
-    'snykNewUnmanagedTest',
-    options,
-  );
+    if (unauthorized) {
+      throw AuthFailedError(error.error, error.code);
```

Comment:
Based on the feedback earlier, I'd suggest just passing in a new error here to satisfy the wrapper. There's not much additional information we'd need for a 401/403 error that we don't already have from the request.

```suggestion
      throw AuthFailedError(new Error("Unauthorized request to unmanaged service"), error.code);
```

Reply:
AuthFailedError requires a string, so I will use a similar approach without the `NewError` call.